omas/omas_physics.py

```diff
@@ -180,6 +181,74 @@ def summary_greenwald(ods, update=True):
     ods_n['summary.global_quantities.greenwald_fraction.value'] = abs(numpy.array(ne_vol_avg) / 1e20 / ip * 1e6 * numpy.pi * a ** 2)
     return ods_n
 
+@add_to__ODS__
+@preprocess_ods('core_profiles', 'equilibrium')
+def summary_lineaverage_density(ods, R1, R2, Z1, Z2, line_grid=2000, time_index=None, update=True):
```

Comment:
The default behavior should be to read interferometer geometry from the ODS. It's in:
`interferometer.channel[:].line_of_sight.first_point.r`
`interferometer.channel[:].line_of_sight.first_point.z`
`interferometer.channel[:].line_of_sight.first_point.phi`
`interferometer.channel[:].line_of_sight.second.r`
...
`interferometer.channel[:].line_of_sight.third_point.r`
...
I think the third point would be the same as the first point for most systems, right?
I guess you could keep the r,z inputs you have as overrides, if you really need them, but getting data from the ODS is probably much smoother.

If you want to simplify the calculation to work just in the R-Z plane (which I assume one would do for the initial implementation), then please add a check that phi matches for all the points. It should probably raise `NotImplementedError` if asked to do the calculation for different phi points unless you bother to support that.

Reply:
`omfit/classes/omfit_omas_d3d.py` already contains `setup_interferometer_hardware_description()`, which loads DIII-D interferometer geometry to the ODS.